torch/lib/THD/base/init_methods/InitMethodTCP.cpp

```diff
@@ -35,8 +35,11 @@ std::string getRandomString()
 
   int fd;
   unsigned int seed;
+  ssize_t bytes_read;
   SYSCHECK(fd = open("/dev/urandom", O_RDONLY));
-  SYSCHECK(read(fd, &seed, sizeof(seed)));
+  SYSCHECK(bytes_read = read(fd, &seed, sizeof(seed)));
```

Comment:
`rank` don't necessarily have to be specified, so you need these uids to perform automatic rank assignment.

You're right, I'll simplify the code tomorrow.